Automattic/jetpack
# Fix media links for tiled gallery

Fixes #25015
Fixes #16940

#### Changes proposed in this Pull Request:

* This PR updates the editor code for the Tiled Gallery block to _always_ link to the original media URL for the image when the "Link To" setting is set to "Media File".
  - Prior to this change, the Square Tiles and Circles layouts would link to the URL for the _currently displayed_ image, which would be a Photon-resized image for those formats
  - This maintains the existing behaviour for the Tiled Mosaic and Tiled Columns layouts
  - The PR also includes a block deprecation so we won't see block validation errors - we should smoothly switch to the new code

#### Other information:


#### Jetpack product discussion

This is a bugfix, so no major discussion needed.

#### Does this pull request change what data or activity we track or use?

No changes to tracking.

#### Testing instructions:

* Set up a standalone site running WordPress 6.0 (so we have Gutenberg in core), and ensure the Jetpack and Jetpack Beta plugins are installed. (The site should _not_ be a WordPress.com Atomic site -- an ephemeral development site should be fine.)
* Ensure that the site is connected to Jetpack.
* Create a new post
* In the new post, add a Tiled Gallery block, and insert multiple non-square images
* Modify the style to be "Square Tiles" or "Circles", and set the "Link To" dropdown to be "Media File"
* View the markup using the code editor -- you should see the Photon-ised URL for the wrapping `<a>` tag around each image -- it will start with `i[0-2].wp.com` and should include an escaped `resize` URL parameter
* Duplicate the Tiled Gallery, and convert the new duplicate block to a reusable block
* Save and publish the post
* In a new tab, navigate to the post on your site
* Verify that clicking on the images in both galleries links to a square image that has a Photonised URL of the form `i[0-2].wp.com`
* In wp-admin, navigate to _Jetpack_ -> _Jetpack Beta_ and click on "Manage" for the Jetpack plugin
* In the "Feature Branch" field, enter `fix/tiled-gallery-media-links` and click on "Activate" to enable this branch
* Navigate to _Posts_ -> _All Posts_ and click on "Edit" for your post from above
* Ensure you force refresh the editor page to get the updated Javascript
* Verify that you don't see a warning about invalid block content
* Inspect the code for the standard block, and verify that the `href` values for the `<a>` tags around the images reference links on the site itself (i.e. `yoursite.tld/wp-content/uploads/*`), and the links don't include a `resize` URL parameter
* Publish the updates to the post
* Refresh the tab that was viewing the post on the site
* Verify that clicking on the images in the standard and reusable Tiled Galleries opens the full-sized image on the site.

Note that for WordPress.com sites, we don't use Photon directly, so I don't believe this specific replacement should impact those cases. However, we should also test the change on both Simple and Atomic sites via D86002-code and the Jetpack Beta plugin respectively. We should see no change in behaviour on those platforms.
